### PR TITLE
[Security] Remove security thingies

### PR DESCRIPTION
**Pull request by mpysiak**
| Q               | A
|-----------------|-----
| Branch?         | 2.0
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | n/a
| License         | MIT

### CODE REVIEW

File: src/Sylius/Bundle/ShopBundle/EventListener/UserRegistrationListener.php [hunk 1]
```diff
@@ -74,6 +74,6 @@ private function enableAndLogin(ShopUserInterface $user): void
         $this->userManager->persist($user);
         $this->userManager->flush();
 
-        $this->userLogin->login($user, $this->firewallContextName);
+        $this->security->login($user, $this->firewallContextName);
```
Comment (NoResponseMate) [inline src/Sylius/Bundle/ShopBundle/EventListener/UserRegistrationListener.php]:
```suggestion
        $this->security->login($user, null, $this->firewallContextName);
```
Firewall name is the 3rd argument in this method

**Review by Wojdylak — DISMISSED** [review]

File: src/Sylius/Behat/Context/Cli/ChangeAdminPasswordContext.php [hunk 2]
```diff
@@ -30,13 +30,15 @@ final class ChangeAdminPasswordContext implements Context
...
+    private array $input = [];
 
     /** @param UserRepositoryInterface<AdminUserInterface> $adminUserRepository */
     public function __construct(
         KernelInterface $kernel,
-        private UserRepositoryInterface $adminUserRepository,
-        private UserPasswordHasherInterface $userPasswordHasher,
+        private readonly UserRepositoryInterface $adminUserRepository,
+        private readonly UserPasswordHasherInterface $userPasswordHasher,
+
```
Comment (GSadee) [inline src/Sylius/Behat/Context/Cli/ChangeAdminPasswordContext.php]:
```suggestion
```

File: src/Sylius/Bundle/CoreBundle/Migrations/Version20241012081014.php [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+final class Version20241012081014 extends AbstractMigration
+{
+    public function getDescription(): string
+    {
+        return 'Remove locked, expires_at and credentials_expire_at columns from User Model';
+    }
+
+    public function up(Schema $schema): void
+    {
+        // this up() migration is auto-generated, please modify it to your needs
```
Comment (GSadee) [inline src/Sylius/Bundle/CoreBundle/Migrations/Version20241012081014.php]:
```suggestion
```
and the same with the others

**Review by GSadee — COMMENTED** [review]
Add notes about changes (security config, removed classes, removed fields in entity, etc) in the UPGRADE file

File: src/Sylius/Component/User/Security/Checker/EnabledUserChecker.php [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+use Symfony\Component\Security\Core\User\UserCheckerInterface;
+use Symfony\Component\Security\Core\User\UserInterface;
+
+final class EnabledUserChecker implements UserCheckerInterface
+{
+    public function checkPreAuth(UserInterface $user): void
+    {
+        if (!$user instanceof ToggleableInterface) {
+            return;
+        }
```
Comment (NoResponseMate) [inline src/Sylius/Component/User/Security/Checker/EnabledUserChecker.php]:
```suggestion
use Symfony\Component\Security\Core\User\UserInterface as SymfonyUserInterface;

final class EnabledUserChecker implements UserCheckerInterface
{
    public function checkPreAuth(UserInterface $user): void
    {
        if (!$user instanceof UserInterface) {
            return;
        }
```
It should only work on sylius user instances

File: src/Sylius/Component/User/tests/Security/Checker/EnabledUserCheckerTest.php
Comment (NoResponseMate) [inline src/Sylius/Component/User/tests/Security/Checker/EnabledUserCheckerTest.php:1]: Happy paths are missing

File: src/Sylius/Bundle/UserBundle/SyliusUserBundle.php [hunk 5]
```diff
@@ -31,8 +30,6 @@ public function getSupportedDrivers(): array
     public function build(ContainerBuilder $container): void
```
Comment (NoResponseMate) [inline src/Sylius/Bundle/UserBundle/SyliusUserBundle.php]: This method can be removed

File: config/packages/security.yaml [hunk 6]
```diff
@@ -41,6 +42,7 @@ security:
         new_api_admin_user:
             pattern: "%sylius.security.new_api_admin_regex%/.*"
             provider: sylius_api_admin_user_provider
+            user_checker: security.user_checker.chain.new_api_admin_user
```
Comment (NoResponseMate) [inline config/packages/security.yaml:45]: This is not related to this PR, but we should change firewall names in the API context,
preferably dropping the `new` and `user` parts

Comment (GSadee) [reply config/packages/security.yaml:45]: I thought the same thing at the first review of this PR 👍🏻

Comment (GSadee) [reply config/packages/security.yaml:45]: #17240

**Review by Wojdylak — APPROVED** [review]

**Review by NoResponseMate — APPROVED** [review]

**Comment by NoResponseMate** [comment]
Thank you, @mpysiak and @GSadee!

**Pull request merged**